### PR TITLE
close connections in finally block

### PR DESCRIPTION
**Pull request by rjullman**

### CODE REVIEW

**Comment by mwakerman** [comment]
Found while trying to write perf-testing CLI and can confirm that this fixed the issue.

**Comment by clockfort** [comment]
I think there's a thing in PG SqlUtils to do this already, but you might want to just write a helper method that does close + catch / log close exception. (looks like there's similar result set and prepared statement closing stuff in BasicSQL, might just add a connection closer helper there?)

Other than that lgtm

File: commons-db/src/main/java/com/palantir/nexus/db/sql/SqlConnectionImpl.java [hunk 1]
```diff
@@ -34,198 +36,513 @@ public SqlConnectionImpl(Supplier<Connection> connectionSupplier, SqlConnectionH
...
+            } catch (SQLException e) {
+                SqlLoggers.LOGGER.error("failed to close sql connection", e);
+            }
+        }
     }
 
     @Override
     public Connection getUnderlyingConnection() {
-        return connectionSupplier.get();
+        Connection conn = connectionSupplier.get();
```
Comment (mwakerman) [inline commons-db/src/main/java/com/palantir/nexus/db/sql/SqlConnectionImpl.java]: You don't want to close this one, it returns the connections.

**Comment by jboreiko** [comment]
Issue already fixed -> https://github.com/palantir/atlasdb/pull/606.  As @rjullman pointed out this would cause problems for using this with a connectionBackedSqlConnection as it would close the underlying connection when it could be used for additional calls resulting in an error.

**Pull request closed without merging**